libnetwork/drivers/overlay/overlay.go

```diff
@@ -29,27 +29,30 @@ const (
 var _ discoverapi.Discover = (*driver)(nil)
 
 type driver struct {
-	bindAddress, advertiseAddress netip.Addr
-
-	config   map[string]interface{}
-	peerDb   peerNetworkMap
-	secMap   *encrMap
-	networks networkTable
-	initOS   sync.Once
-	keys     []*key
-	peerOpMu sync.Mutex
-	mu       sync.Mutex
+	// Immutable; mu does not need to be held when accessing these fields.
+
+	config map[string]interface{}
+	initOS sync.Once
+
+	// Has dedicated locks; use the locks inside the structs when accessing them.
+
+	secMap encrMap
```

Comment:
I guess this now applies to `secMap` (not the other things further down), so we may as well make it a "godoc" for that field;

```suggestion
	// secMap has dedicated locks; use the locks inside the structs when accessing them.
	secMap encrMap
```

Reply:
The comment applies to all fields under the section-heading comment. There just happens to only be one field in the section at the moment. A new section heading comment "clears" the preceding comments. Nobody is going to be confused about the immutability of secMap just because there is a comment about immutability some lines above.